src/samVisualize/commands/visualizeSamTemplate/samVisualization.ts

```diff
@@ -0,0 +1,227 @@
+
+            const graphObjectString = JSON.stringify(graphObject)
+
+            const primaryResourceListString = readFileSync(
+                this.extensionContext.asAbsolutePath(
+                    join('resources', 'light', 'samVisualize', 'primaryResources.json')
+                )
+            ).toString()
+
+            const iconsDir = this.extensionContext.asAbsolutePath(join('resources', 'light', 'samVisualize', 'icons'))
+
+            const iconsDataString = JSON.stringify(generateIconsMap(iconsDir, panel.webview))
+
+            const webviewStylesheetUri = panel.webview.asWebviewUri(
+                vscode.Uri.file(this.extensionContext.asAbsolutePath(join('media', 'css', 'samVisualize.css')))
+            )
+
+            const webviewBodyScriptUri = panel.webview.asWebviewUri(
+                // Change those file names
+                vscode.Uri.file(
+                    this.extensionContext.asAbsolutePath(join('samVisualizeRenderJS', 'samVisualizeRenderBundle.js'))
+                )
+            )
+            panel.webview.html = this.getWebviewContent(
+                graphObjectString,
+                primaryResourceListString,
+                iconsDataString,
+                webviewStylesheetUri,
+                webviewBodyScriptUri,
+                panel.webview.cspSource
+            )
+        } catch {}
+
+        // To close the visualization if the associated template no longer exists
+        this.disposables.push(
+            vscode.workspace.onDidCloseTextDocument(e => {
+                if (!this.templateDoesExist() && this.webviewPanel) {
+                    panel.dispose()
+                    vscode.window.showErrorMessage('Template associated with visualization closed or renamed')
+                }
+            })
+        )
+
+        const debounceInterval = 1000
+        const debouncedUpdate = _.debounce(this.updateVisualization.bind(this), debounceInterval)
+
+        // Update the visualization on template change
+        this.disposables.push(
+            vscode.workspace.onDidChangeTextDocument(async changedTextDocument => {
+                if (changedTextDocument.document.uri.path === this.textDocumentUri.path) {
+                    debouncedUpdate(changedTextDocument.document.getText())
+                }
+            })
+        )
+
+        const decorationType = vscode.window.createTextEditorDecorationType({
+            backgroundColor: 'rgb(135,206,250,0.3)', // Light sky blue
+        })
+
+        // Handle click events from the webview
+        this.disposables.push(
+            panel.webview.onDidReceiveMessage(message => {
+                // Reveal the template to the left of the webview, unless the webview is in the first column.
+                // In this case, reveal to the right. This way, the template will also reveal beside the webview.
+                const columnToShow =
+                    this.webviewPanel!.viewColumn === vscode.ViewColumn.One
```

Comment:
Does `beside` not work? https://code.visualstudio.com/api/references/vscode-api#ViewColumn

Reply:
Since the webview is rendered `beside` the template, the familiar arrangement will be template on the left, and webview on the right. If we reveal the template `beside`, it'll appear to the right of webview, which is not desirable, as three viewcolumns will be open. The only way a webview gets into the first column is if it's moved there, and I thought it would be best to reveal the template to the right in this case only, so that we stay as consistent with the "template on the left, webview on the right" unless the customer purposefully moves the webview to the first column (in which case they still have access to the template, without moving the webview column).